README.md

```diff
@@ -133,6 +133,78 @@ fn main() {
 }
 ```
 
+## Example: Futures + TLS
+
+By default, tarpc uses a `TcpStream` for communication, however you can also opt to use a
+`TlsStream<TcpStream>` when the `tls` feature of `tarpc` is enabled.
+
+When using TLS, some additional information is required. You will need to make [`TlsAcceptor`] and
+`TlsClientContext` structs; `TlsClientContext` requires a [`TlsConnector`]. The [`TlsAcceptor`] and
+[`TlsConnector`] structs are defined in the [native-tls] crate which is exposed by `tarpc`.
```

Comment:
Yes, that's the direction I was going. I've seen libraries that have public interfaces which require structs from other crates generally do this. You can see in [reqwest](https://github.com/seanmonstar/reqwest/blob/master/src/lib.rs#L100) and [hyper](https://github.com/hyperium/hyper/blob/master/src/lib.rs#L36). I think the best approach would be to just re-export only the external types we need, not the whole crate. I'll make that change when I do my updates.